hooks/useClickOutside.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+/**
+ * @typedef {Object} ElementRef
+ * @property {myRef} - reference to the DOM element
+ * @property {wasClicked} - state
+ */
+
+/**
+ *
+ * Custom hook which return an object with the refence to the DOM element,
+ * the state and the setter of the state.
+ * @param {boolean} initialValue Initial value of the state.
+ * @return {ElementRef} Object containing the coordinates of the mouse pointer.
+ */
+const useClickOutside = (initialValue) => {
+  const [wasClicked, setWasClicked] = React.useState(initialValue);
```

Comment:
```suggestion
  const [wasClicked, setWasClicked] = React.useState(false);
```

I suggest keeping this false. I don't see a use case for accepting an initial value